### PR TITLE
Enable doctests for GroupBy methods

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR enables doctests for some GroupBy methods that are not currently tested due to not meeting the inclusion criteria in our doctest class. This includes enabling tests for `GroupBy.apply` with `engine='jit'`. 

came up during https://github.com/rapidsai/cudf/pull/11452

### CODE REVIEW

File: python/cudf/cudf/core/groupby/groupby.py [hunk 1]
```diff
@@ -1652,28 +1676,6 @@ def fillna(
...
-            **groupby.fillna**
-
-            This function may return result in different format to the method
-            Pandas supports. For example:
-
-            .. code-block::
-
-                >>> df = pd.DataFrame({'k': [1, 1, 2], 'v': [2, None, 4]})
-                >>> gdf = cudf.from_pandas(df)
-                >>> df.groupby('k').fillna({'v': 4}) # pandas
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/groupby/groupby.py:1666]: AFAICT this is no longer the behavior of pandas and `reset_index` must be called somewhere in pandas internally.

```python
>>> import pandas as pd
>>> pd.__version__
'1.5.3'
>>> df = pd.DataFrame({'k': [1, 1, 2], 'v': [2, None, 4]})
>>> df.groupby('k').fillna({'v': 4})
     v
0  2.0
1  4.0
2  4.0
```

File: python/cudf/cudf/core/groupby/groupby.py [hunk 2]
```diff
@@ -430,6 +433,12 @@ def agg(self, func):
...
 
+        >>> import cudf
+        >>> a = cudf.DataFrame({
+        ...     'a': [1, 1, 2],
+        ...     'b': [1, 2, 3],
+        ...     'c': [2, 2, 1]
+        ... })
         >>> a.groupby('a').agg(['sum', 'min'])
             b       c
           sum min sum min
```
Comment (galipremsagar) [inline python/cudf/cudf/core/groupby/groupby.py:444]:
```suggestion
                b       c
          sum min sum min
```

While we are at it, can we properly indent the first lines of all the column names in this file? These are translating to improper indentation on the actual pages too, for example: https://docs.rapids.ai/api/cudf/nightly/api_docs/api/cudf.core.groupby.groupby.groupby.agg#cudf.core.groupby.groupby.GroupBy.agg

File: python/cudf/cudf/core/groupby/groupby.py [hunk 3]
```diff
@@ -1218,8 +1240,10 @@ def describe(self, include=None, exclude=None):
...
         --------
         >>> import cudf
-        >>> gdf = cudf.DataFrame({"Speed": [380.0, 370.0, 24.0, 26.0],
-                                  "Score": [50, 30, 90, 80]})
+        >>> gdf = cudf.DataFrame({
+        ...     "Speed": [380.0, 370.0, 24.0, 26.0],
+        ...      "Score": [50, 30, 90, 80],
+        ... })
         >>> gdf
         Speed  Score
```
Comment (galipremsagar) [inline python/cudf/cudf/core/groupby/groupby.py:1248]:
```suggestion
          Speed  Score
```

While we are at it, can we also indent the column names properly in this file? These are translating to improperly formatted output in the docs, for example: https://docs.rapids.ai/api/cudf/nightly/api_docs/api/cudf.core.groupby.groupby.dataframegroupby.describe

Comment (brandon-b-miller) [reply python/cudf/cudf/core/groupby/groupby.py]: Fixed in the areas I spotted (3), let me know if I missed any here.

**Pull request merged**